src/test/storage/group_key_index_test.cpp

```diff
@@ -11,15 +11,32 @@
 #include "storage/base_segment.hpp"
 #include "storage/chunk.hpp"
 #include "storage/index/group_key/group_key_index.hpp"
+#include "storage/value_segment.hpp"
 #include "types.hpp"
 
 namespace opossum {
 
 class GroupKeyIndexTest : public BaseTest {
  protected:
   void SetUp() override {
-    dict_segment = BaseTest::create_dict_segment_by_type<pmr_string>(
-        DataType::String, {"hotel", "delta", "frank", "delta", "apple", "charlie", "charlie", "inbox"});
+    auto value_segment_str = std::make_shared<ValueSegment<pmr_string>>(true);
+    //                                      //  position
+    value_segment_str->append(NULL_VALUE);  //  0
+    value_segment_str->append("hotel");     //  1
+    value_segment_str->append("delta");     //  2
+    value_segment_str->append("frank");     //  3
+    value_segment_str->append("delta");     //  4
+    value_segment_str->append(NULL_VALUE);  //  5
+    value_segment_str->append(NULL_VALUE);  //  6
+    value_segment_str->append("apple");     //  7
+    value_segment_str->append("charlie");   //  8
+    value_segment_str->append("charlie");   //  9
+    value_segment_str->append("inbox");     // 10
+    value_segment_str->append(NULL_VALUE);  // 11
```

Comment:
Let this be reflective of the example in the hpp? Also, the example is missing the offset to the end of the postings list. That was broken before.

Reply:
https://github.com/hyrise/hyrise/pull/1772/commits/068db8c262ffea70e14801ccdbfb89f00f5085b5